my_sprites.py

```diff
@@ -35,9 +39,13 @@ def __init__(
 
         # hp
         self._max_hp = max_hp
-        self.cur_hp = max_hp
+        self._hp = max_hp
 
+        self.window = window
         self.speed = speed
+        self.target = target
+        self.roaming_dist = roaming_dist
+        self.state = EnemyState.ROAMING
```

Comment:
Please add state to the arguments in init(). This state can be the default value.